.github/workflows/ci.yml

```diff
@@ -27,56 +28,60 @@ on:
   # Triggers when a push is made to the master branch (either by
   # merging a pull request, or by direct commit)
   push:
-    branches:
-      - master
+    branches: [ master ]
 
   # Run manually from the Actions tab
   workflow_dispatch:
 
 # ------------------------------ JOBS ------------------------------ #
-# This workflow contains a single job called `testing`, which compiles
-# the AtChem2 model, runs the Testsuite, and checks the Testsuite code
-# coverage using Codecov
+# This workflow contains a single job called `ci_testing` which
+# compiles the AtChem2 model, runs the Testsuite, and checks the code
+# coverage using Codecov (https://app.codecov.io/gh/AtChem/AtChem2)
 jobs:
-  testing:
+  ci_testing:
     # The job runs on both linux (ubuntu) and macos runner images,
     # with three versions of gnu fortran -- the runner images are
     # described here: https://github.com/actions/runner-images
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [ubuntu-22.04, macos-11] #, macos-12]
+        os: [ubuntu-20.04, ubuntu-22.04, macos-11, macos-12]
         fortran: [9, 10, 11]
-        # exclude: #TODO: fortran9 does not work on macos12
-        # - os: macos-12
-        #   fortran: 9
+        exclude: # TODO: gfortran-9 does not work on macos-12
+        - os: macos-12
+          fortran: 9
       fail-fast: false
 
     # -------------------------------------------------------------
-    # Sequence of tasks to be executed as part of the `testing` job:
-    # 1. checkout the repository
-    # 2. install gfortran, AtChem2 dependencies
-    # 3. compile AtChem2
-    # 4. run all tests (indent, style, unit, model)
-    # 5. recompile AtChem2, run unit and model tests for Codecov
+    # Sequence of tasks to be executed as part of the `ci_testing` job:
+    # 1. Checkout the repository
+    # 2. Install gfortran, AtChem2 dependencies
+    # 3. Compile AtChem2
+    # 4. Run the Testsuite with default settings
+    # 5. Recompile AtChem2, run unit and model tests for Codecov
     steps:
 
       # -------------------------------------------------------------
       # (1) Checkout the repository under $GITHUB_WORKSPACE
-      - uses: actions/checkout@v3
+      - name: Checkout repo
+        uses: actions/checkout@v4
         with:
           fetch-depth: 2
```

Comment:
@spco is `with: fetch-depth: 2` necessary here? I can't understand if it does something in the context of this workflow.

Reply:
It reduces the number of commits fetched down to 2. By default, the full history is fetched, which is unnecessary here and bloats things. So it just speeds things up a bit - particularly helpful when we have a large number of large files that have gone through repeated revisions (like the output files). It'll function fine without this line, if you prefer clarity or succinctness.

Reply:
Ah okay. The documentation was a bit confusing to me, but this makes sense. Thanks